packages/api/src/car.js

```diff
@@ -124,3 +171,34 @@ export async function sizeOf (response) {
   }
   return size
 }
+
+/**
+ * Returns the DAG size of the CAR but only if the graph is complete.
+ * @param {Blob} car
+ */
+async function getDagSize (car) {
+  const decoders = [pb, raw, cbor]
+  const bytes = new Uint8Array(await car.arrayBuffer())
+  const reader = await CarReader.fromBytes(bytes)
+  const rootCid = (await reader.getRoots())[0]
+
+  const getBlock = async cid => {
+    const rawBlock = await reader.get(cid)
+    if (!rawBlock) throw new Error(`missing block for ${cid}`)
+    const { bytes } = rawBlock
+    const decoder = decoders.find(d => d.code === cid.code)
+    if (!decoder) throw new Error(`missing decoder for ${cid.code}`)
+    return new Block({ cid, bytes, value: decoder.decode(bytes) })
+  }
+
+  const getSize = async cid => {
```

Comment:
aww man, we really need a lightwight unixfs decoder implementation as for most dags this could just be "decode the root and inspect the size property"

Reply:
the old dag-pb would have given us this out of the box! and who uses dag-pb for anything other than unixfs! I'd like to take this up with the manager.